misc/v3-task-endpoint-validator-windows/v3-task-endpoint-validator-windows.go

```diff
@@ -0,0 +1,387 @@
+	if err != nil {
+		return err
+	}
+
+	seelog.Infof("Received container stats: %s \n", string(body))
+
+	return nil
+}
+
+func verifyTaskStats(client *http.Client, taskStatsEndpoint string) error {
+	body, err := metadataResponse(client, taskStatsEndpoint)
+	if err != nil {
+		return err
+	}
+
+	seelog.Infof("Received task stats: %s \n", string(body))
+
+	return nil
+}
+
+func verifyTaskMetadataResponse(taskMetadataRawMsg json.RawMessage) error {
+	var err error
+	taskMetadataResponseMap := make(map[string]json.RawMessage)
+	json.Unmarshal(taskMetadataRawMsg, &taskMetadataResponseMap)
+
+	if err = fieldEqual(taskMetadataResponseMap, "Cluster", "ecs-functional-tests"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(taskMetadataResponseMap, "TaskARN"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(taskMetadataResponseMap, "Family"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(taskMetadataResponseMap, "Revision", "1"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(taskMetadataResponseMap, "DesiredStatus", "RUNNING"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(taskMetadataResponseMap, "KnownStatus", "RUNNING"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(taskMetadataResponseMap, "PullStartedAt"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(taskMetadataResponseMap, "PullStoppedAt"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(taskMetadataResponseMap, "Containers"); err != nil {
+		return err
+	}
+
+	var containersMetadataResponseArray []json.RawMessage
+	json.Unmarshal(taskMetadataResponseMap["Containers"], &containersMetadataResponseArray)
+
+	return verifyContainerMetadataResponse(containersMetadataResponseArray[0])
+}
+
+func verifyContainerMetadataResponse(containerMetadataRawMsg json.RawMessage) error {
+	var err error
+	containerMetadataResponseMap := make(map[string]json.RawMessage)
+	json.Unmarshal(containerMetadataRawMsg, &containerMetadataResponseMap)
+
+	if err = fieldNotEmpty(containerMetadataResponseMap, "DockerId"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(containerMetadataResponseMap, "Name", "v3-task-endpoint-validator-windows"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(containerMetadataResponseMap, "DockerName"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(containerMetadataResponseMap, "Image"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(containerMetadataResponseMap, "Image", "amazon/amazon-ecs-v3-task-endpoint-validator-windows"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(containerMetadataResponseMap, "ImageID"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(containerMetadataResponseMap, "DesiredStatus", "RUNNING"); err != nil {
+		return err
+	}
+
+	if err = fieldEqual(containerMetadataResponseMap, "KnownStatus", "RUNNING"); err != nil {
+		return err
+	}
+
+	if err = fieldNotEmpty(containerMetadataResponseMap, "Limits"); err != nil {
```

Comment:
Instead of having the condition check repeated, you can simply have a list of strings with all the expected keys and do something like: 
```
for _, key := range keys {
    if err = fieldEqual(containerMetadataResponseMap, key); err != nil {
        return err
    }
}
```
This would make it easier to add more fields to these tests.

Reply:
Similarly for `fieldEqual` checks, you can have a map of expected values too